### PR TITLE
fix logpow for case x = m = 0

### PR DESCRIPTION
**Pull request by katosh**
This should fix the issues discussed here: https://discourse.pymc.io/t/multinomial-returns-nan/1413/2

### CODE REVIEW

File: pymc3/distributions/multivariate.py [hunk 1]
```diff
@@ -383,10 +383,9 @@ class Dirichlet(Continuous):
 
     .. math::
 
-       f(\mathbf{x}) =
-           \frac{\Gamma(\sum_{i=1}^k \theta_i)}{\prod \Gamma(\theta_i)}
-           \prod_{i=1}^{k-1} x_i^{\theta_i - 1}
-           \left(1-\sum_{i=1}^{k-1}x_i\right)^\theta_k
+       f(\mathbf{x}|\mathbf{a}) =
+           \frac{\Gamma(\sum_{i=1}^k a_i)}{\prodi_{i=1}^k \Gamma(a_i)}
```
Comment (gBokiau) [inline pymc3/distributions/multivariate.py]: `\prodi_{i=1}`: ~~you're missing a space there~~ there shouldn't be an i there (right after \prod)

Comment (katosh) [reply pymc3/distributions/multivariate.py]: You are right! Thank you for checking that! Fixed it in the commit below.

**Comment by gBokiau** [comment]
Hey, wait… I thought how did I not see this, but the way it was implemented already solved this, since 0*log(0) is also 0 (at least for theano). And I think the way it was is more effective computationally.

So https://github.com/pymc-devs/pymc3/pull/3055 was enough, IMO.

**Comment by katosh** [comment]
The examples
```
import numpy as np
import pymc3 as pm
np.exp(pm.Multinomial.dist(n = 1, p = np.array([0, 1])).logp(np.array([0, 1])).eval())
np.exp(pm.Dirichlet.dist(a = np.array([2, 1])).logp(np.array([1, 0])).eval())
```
both return `nan` in the current master and the correct results (1 and 2) after these changes. Maybe it behaved differently if the given parameter where theano tensors?

**Comment by gBokiau** [comment]
Ah, right, because they're vectors and tt.eq isn't elemwise, makes sense!
My fault, I should have checked. Thanks!

**Comment by gBokiau** [comment]
Oh, no, `tt.eq(x, 0)` *is* elemwise, so it should have worked. The odd thing is  `0 * tt.log(0)` returns `0` but `[0,y]*tt.log([0,y])` returns `[nan, y*log(y)]`. @nouiz is that expected behaviour?

**Comment by katosh** [comment]
From a mathematical point of view it is not easy to say what `0 * -inf` should return. Nicely explained here: https://math.stackexchange.com/questions/28940/why-is-infinity-multiplied-by-zero-not-an-easy-zero-answer
So we could even question the scalar case `0 * tt.log(0)`. Numpy decided to reurn `nan` for `0*np.inf`.

**Comment by gBokiau** [comment]
I agree, just wondering why the implementation is inconsistent in Theano. Things being what they are though, the PR has my full support ;)

**Comment by junpenglao** [comment]
Thanks for the PR @katosh (congrats on the first contribution), and thanks for the comments @gBokiau ;-)

**Pull request merged**